libfwupdplugin/fwupdplugin.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2015 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#pragma once
+
+/**
+ * SECTION:fwupdplugin
+ * @short_description: Helper objects for plugins interacting with fwupd daemon
+ */
+
+#define __FWUPDPLUGIN_H_INSIDE__
+
+#include <libfwupdplugin/fu-archive.h>
+#include <libfwupdplugin/fu-chunk.h>
+#include <libfwupdplugin/fu-common.h>
+#include <libfwupdplugin/fu-common-cab.h>
+#include <libfwupdplugin/fu-common-guid.h>
+#include <libfwupdplugin/fu-common-version.h>
+#include <libfwupdplugin/fu-device.h>
+#include <libfwupdplugin/fu-device-locker.h>
+#include <libfwupdplugin/fu-device-metadata.h>
+#include <libfwupdplugin/fu-dfu-firmware.h>
+#include <libfwupdplugin/fu-firmware.h>
+#include <libfwupdplugin/fu-firmware-common.h>
+#include <libfwupdplugin/fu-firmware-image.h>
+#include <libfwupdplugin/fu-hwids.h>
+#include <libfwupdplugin/fu-history.h>
+#include <libfwupdplugin/fu-ihex-firmware.h>
+#include <libfwupdplugin/fu-io-channel.h>
+#include <libfwupdplugin/fu-plugin.h>
+#include <libfwupdplugin/fu-plugin-vfuncs.h>
+#include <libfwupdplugin/fu-progressbar.h>
```

Comment:
I don't think this should be exported in fwupdplugin.so - I think it probably out to be a static library built in src and just link_with'd in the places that need it. Maybe fwupdnoinst.a or something.

Reply:
What do you think about splitting it up to a device attribute for percentage and using the existing device status attribute? Then we can keep it only as part of daemon and tools.

Reply:
I think the status property makes a lot of sense, although I don't know how hard it is to have multiple FuDevices updating one progressbar. I think the issue is also that other stuff uses it, e.g. plugins/dfu/dfu-tool.c -- although we could just nuke it from there as the DFU tool is less and less important to me given all the recent changes.